actions/setup-nix/action.yml

```diff
@@ -1,107 +1,159 @@
-    description:
-      "grafana org/tenant id - required if metrics-job-name is passed"
+      "Private signing key for Nix cache (only required if pushing to cache)"
     required: false
+    default: ""
 
 runs:
   using: composite
   steps:
-    - name: configure AWS credentials
+    - name: 🔐 Configure AWS credentials (if enabled)
       if: inputs.enable-aws == 'true'
-      uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+      uses: aws-actions/configure-aws-credentials@050bf7aae915badb82daa8e68fd95c9070706953 # v4.0.2
       with:
         role-to-assume: ${{ inputs.role-to-assume }}
         role-duration-seconds: ${{ inputs.role-duration-seconds }}
         aws-region: ${{ inputs.aws-region }}
         mask-aws-account-id: true
 
-    - name: setup AWS credentials for nix
+    - name: 🛠️ Setup AWS credentials for Nix (if enabled)
       if: inputs.enable-aws == 'true'
       shell: bash
       run: |
-        # set up folder with AWS configs
-        # nix reads AWS access from folder config - does not work with env vars
+        echo "🔑 Setting up AWS credentials for Nix..."
         aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
         aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
         aws configure set aws_session_token $AWS_SESSION_TOKEN
         aws configure set region ${{ inputs.aws-region }}
 
-        # determine root path
         ROOT_PATH=""
         if [[ "${{ runner.os }}" == "Linux" ]]; then
           ROOT_PATH="/root"
         elif [[ "${{ runner.os }}" == "macOS" ]]; then
           ROOT_PATH="/var/root"
         else
-          echo "${{ runner.os }} not supported"
+          echo "⚠️ ${ { runner.os }} not supported!"
           exit 1
         fi
 
-        # symlink credential folders for nix build access
-        # nix build is run via daemon on root, credentials linked for daemon access
+        echo "🔗 Symlinking AWS credentials for Nix daemon..."
         sudo ln -s ~/.aws $ROOT_PATH
 
-    - uses: cachix/install-nix-action@ba0dd844c9180cbf77aa72a116d6fbc515d0e87b # v27
+    - name: 💾 Store private signing key (if provided)
+      if: ${{ inputs.private-signing-key != '' }}
+      shell: bash
+      run: |
+        echo "🔑 Storing private signing key for cache signing..."
+        echo "${{ inputs.private-signing-key }}" > ${{ github.workspace }}/nix-signing.private
+        chmod 600 ${{ github.workspace }}/nix-signing.private
+
+    - name: Preprocess Public Keys
+      shell: bash
+      run: |
+        # Preprocess the public keys and store them in a variable
+        PUBLIC_KEYS=$(echo "${{ inputs.cache-pubkeys }}" | tr ',' ' ')
+        echo "Processed Public Keys: $PUBLIC_KEYS"
+
+        # Store the processed public keys in an environment variable for use in later steps
+        echo "PUBLIC_KEYS=$PUBLIC_KEYS" >> $GITHUB_ENV
+
+    - uses: DeterminateSystems/nix-installer-action@da36cb69b1c3247ad7a1f931ebfd954a1105ef14 # v14
```

Comment:
v14?

I do feel we could/should extract the install action (third-party) from the cache setup action (our S3 setup). Would be cleaner, and users could configure with custom `install_url` etc.

Reply:
One issue I see here is that the cache setup paths might change depending on the gh action used to install nix. For example I'm not 100% sure if all installer place configs under `/etc/nix/nix.conf`.It might make it a bit harder to support all possible nix installers. But I agree it would make the params a bit cleaner in general.

Reply:
As an idea, there is an $NIX_CONF_DIR env var that should be available.

https://nix.dev/manual/nix/2.24/command-ref/conf-file.html

Reply:
I was able to separate the install / cache action now https://github.com/smartcontractkit/.github/pull/502/commits/f903c07a199c1b28a8256c7e10b4c2d2a45f6db7

For now im defaulting to `/etc/nix/nix.conf` as the `NIX_CONF_DIR` appears to be empty even after installing Nix. Let me know your thoughts with this new separations!